Upstream-Status: Backport

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From dc6468b038823e8fe1e69f9771c336ac65424aa9 Mon Sep 17 00:00:00 2001
From: Nick Diego Yamane <nickdiego@igalia.com>
Date: Sat, 16 Mar 2019 19:02:43 +0000
Subject: [PATCH 09/27] ozone/wayland: Clean up WaylandCursor

This CL makes a code cleanup in WaylandCursor class:
 - Remove unneeded/unused header includes
 - Switch to forward declarations instead of header includes
 where possible
 - Fix lint errors

Bug: 578890
Change-Id: I481723749b1c7ddb5dcd7d4569d3517be8a924ec
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1525104
Reviewed-by: Maksim Sisov <msisov@igalia.com>
Commit-Queue: Nick Yamane <nickdiego@igalia.com>
Cr-Commit-Position: refs/heads/master@{#641452}
---
 ui/ozone/platform/wayland/wayland_cursor.cc | 11 ++++-------
 ui/ozone/platform/wayland/wayland_cursor.h  |  4 +++-
 2 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/ui/ozone/platform/wayland/wayland_cursor.cc b/ui/ozone/platform/wayland/wayland_cursor.cc
index 8c1f4bf5d9f4..725e70cfcf40 100644
--- a/ui/ozone/platform/wayland/wayland_cursor.cc
+++ b/ui/ozone/platform/wayland/wayland_cursor.cc
@@ -4,14 +4,10 @@
 
 #include "ui/ozone/platform/wayland/wayland_cursor.h"
 
-#include <sys/mman.h>
-#include <vector>
-
 #include "base/memory/shared_memory.h"
-#include "third_party/skia/include/core/SkCanvas.h"
+#include "third_party/skia/include/core/SkBitmap.h"
 #include "ui/gfx/skia_util.h"
 #include "ui/ozone/platform/wayland/wayland_connection.h"
-#include "ui/ozone/platform/wayland/wayland_pointer.h"
 #include "ui/ozone/platform/wayland/wayland_util.h"
 
 namespace ui {
@@ -23,15 +19,16 @@ WaylandCursor::~WaylandCursor() = default;
 // static
 void WaylandCursor::OnBufferRelease(void* data, wl_buffer* buffer) {
   auto* cursor = static_cast<WaylandCursor*>(data);
-  DCHECK(cursor->buffers_.count(buffer) > 0);
+  DCHECK_GT(cursor->buffers_.count(buffer), 0u);
   cursor->buffers_.erase(buffer);
 }
 
 void WaylandCursor::Init(wl_pointer* pointer, WaylandConnection* connection) {
   DCHECK(connection);
+  DCHECK(connection->shm());
+  DCHECK(connection->compositor());
 
   input_pointer_ = pointer;
-
   shm_ = connection->shm();
   pointer_surface_.reset(
       wl_compositor_create_surface(connection->compositor()));
diff --git a/ui/ozone/platform/wayland/wayland_cursor.h b/ui/ozone/platform/wayland/wayland_cursor.h
index e7e8368d7233..00de27ee9e84 100644
--- a/ui/ozone/platform/wayland/wayland_cursor.h
+++ b/ui/ozone/platform/wayland/wayland_cursor.h
@@ -6,16 +6,18 @@
 #define UI_OZONE_PLATFORM_WAYLAND_WAYLAND_CURSOR_H_
 
 #include <wayland-client.h>
+
 #include <memory>
 #include <utility>
 #include <vector>
 
 #include "base/containers/flat_map.h"
 #include "base/macros.h"
-#include "third_party/skia/include/core/SkBitmap.h"
 #include "third_party/skia/include/core/SkSurface.h"
 #include "ui/ozone/platform/wayland/wayland_object.h"
 
+class SkBitmap;
+
 namespace base {
 class SharedMemory;
 }
-- 
2.20.1

